src/ruby/lib/grpc/generic/rpc_desc.rb

```diff
@@ -47,43 +47,85 @@ def unmarshal_proc(target)
       proc { |o| unmarshal_class.method(unmarshal_method).call(o) }
     end
 
-    def handle_request_response(active_call, mth)
+    def handle_request_response(active_call, mth, inter_ctx)
       req = active_call.read_unary_request
-      resp = mth.call(req, active_call.single_req_view)
-      active_call.server_unary_response(
-        resp, trailing_metadata: active_call.output_metadata)
+      call = active_call.single_req_view
+
+      inter_ctx.intercept!(
+        :request_response,
+        method: mth,
+        call: call,
+        request: req
+      ) do
+        resp = mth.call(req, call)
+        active_call.server_unary_response(
```

Comment:
Hi @splittingred 
Is there any reason why the active_call is sending the response within the yield block? As of now, yield's return value inside a middleware is nil. Looks like the response is sent as soon as the handler method is invoked and the middlewares cannot even read the response.
But, the design document seemed like we can catch the response in the middlewares. Has the design changed keeping the security aspect in mind?
Also, just to make my understanding clear, the server interceptors as of now can only modify the response metadata. Am I right?